x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/RuleQueryBuilder.java

```diff
@@ -0,0 +1,334 @@
+        Supplier<List<String>> curatedIdSupplier,
+        Supplier<List<Item>> curatedDocsSupplier
+
+    ) {
+        if (organicQuery == null) {
+            throw new IllegalArgumentException("organicQuery must not be null");
+        }
+        if (matchCriteria == null || matchCriteria.isEmpty()) {
+            throw new IllegalArgumentException("matchCriteria must not be null or empty");
+        }
+        for (String matchCriteriaKey : matchCriteria.keySet()) {
+            if (ALLOWED_MATCH_CRITERIA.contains(matchCriteriaKey) == false) {
+                throw new IllegalArgumentException("matchCriteria key [" + matchCriteriaKey + "] is not allowed");
+            }
+        }
+        if (rulesetIds == null || rulesetIds.isEmpty()) {
+            throw new IllegalArgumentException("rulesetIds must not be null or empty");
+        }
+
+        this.organicQuery = organicQuery;
+        this.matchCriteria = matchCriteria;
+        this.rulesetIds = rulesetIds;
+        this.curatedIds = curatedIds;
+        this.curatedIdSupplier = curatedIdSupplier;
+        this.curatedDocs = curatedDocs;
+        this.curatedDocsSupplier = curatedDocsSupplier;
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        if (curatedIdSupplier != null) {
+            throw new IllegalStateException("curatedIdSupplier must be null, can't serialize suppliers, missing a rewriteAndFetch?");
+        }
+        if (curatedDocsSupplier != null) {
+            throw new IllegalStateException("curatedDocsSupplier must be null, can't serialize suppliers, missing a rewriteAndFetch?");
+        }
+
+        out.writeNamedWriteable(organicQuery);
+        out.writeGenericMap(matchCriteria);
+        out.writeStringCollection(rulesetIds);
+        if (curatedIds != null) {
+            out.writeBoolean(true);
+            out.writeStringCollection(curatedIds);
+        } else {
+            out.writeBoolean(false);
+        }
+        if (curatedDocs != null) {
+            out.writeBoolean(true);
+            out.writeList(curatedDocs);
+        } else {
+            out.writeBoolean(false);
+        }
+    }
+
+    public List<String> rulesetIds() {
+        return rulesetIds;
+    }
+
+    public Map<String, Object> matchCriteria() {
+        return matchCriteria;
+    }
+
+    public QueryBuilder organicQuery() {
+        return organicQuery;
+    }
+
+    @Override
+    protected void doXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(NAME);
+        builder.field(ORGANIC_QUERY_FIELD.getPreferredName(), organicQuery);
+        builder.startObject(MATCH_CRITERIA_FIELD.getPreferredName());
+        builder.mapContents(matchCriteria);
+        builder.endObject();
+        builder.startArray(RULESET_IDS_FIELD.getPreferredName());
+        for (String rulesetId : rulesetIds) {
+            builder.value(rulesetId);
+        }
+        builder.endArray();
```

Comment:
You need to call  `printBoostAndQueryName(builder);` here to ensure that `_name` and `boost` are part of the serialisation. That's the reason why the serialisation tests are failing.

Reply:
🤦‍♀️ Thank you - I knew it was going to be something very simple!